xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/edit.vm

```diff
@@ -139,7 +139,7 @@
   #end
   #template("header.vm")
   #if($editor == 'wiki' || $editor == 'wysiwyg')
-    <form id="edit" method="post" action="$doc.getURL('preview')" class="withLock form">
+    <form id="edit" method="post" action="$doc.getURL('preview')" class="withLock form xform">
```

Comment:
I'd like to know why we need to keep the ``form`` class. Would be good if you can check:
* if there are CSS styles associated with it
* if there is JavaScript code that uses this CSS class

If not then we should remove it. It feels wrong to have both ``form`` and ``xform`` but we need to understand what we loose / break if we remove the ``form`` CSS class.

Reply:
I looked through the git history of this file, and the `form` CSS class is present from the beginning in the `edit.vm`, but it has not been present in the `xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/edit.vm` where this template originally came from.

One can do a `find . -name '*.css' | xargs fgrep '.form'` to find if this CSS class is used in any CSS file (from within a running XWiki webapp; the source tree also has `*.less` files from which the CSS files are generated. I guess `.css.vm` files are no longer around.)
I am not sure that the same approach works for JavaScript files, however. There are different ways to get at an element which has a CSS class `form`, and just grepping for `form` will create too many hits.
Maybe the WYSIWYG editor stops working, or the Wiki editor, if this class is omitted? If it is not, I feel the CSS class is unused.

The initial commit introducing the `form` CSS class is [dcd85167489885e2017e4c56f6a4768924542927](https://github.com/xwiki/xwiki-platform/commit/dcd85167489885e2017e4c56f6a4768924542927) and has been done by @vmassol but the commit is way too large for me to see why this CSS class might have been introduced.

Reply:
@ClemensRobbenhaar Big thanks for this!

Reply:
Yes, the WYSIWYG and Wiki editors are working fine. I believe `form` class is not being used, a hint I took from `editclass` and `editobject` .vm files which have `withLock xform` instead of `withLock form`.
I tested `withLock xform` and `withLock form xform` , they have no difference and both work fine.

Reply:
@Haxsen OK, then remove the ``form`` CSS class.

Reply:
Done.